### PR TITLE
elv-apps-projects#234 - Content admin permissions

### PR DESCRIPTION
**Pull request by elv-kevin**
https://github.com/qluvio/elv-apps-projects/issues/234

### CODE REVIEW

**Review by elv-zenia — APPROVED** [review]

File: src/clients/Fabric.js [hunk 1]
```diff
@@ -1088,6 +1088,35 @@ const Fabric = {
 
   /* Object creation / modification */
 
+  GetContentAdminsGroupAddress: async () => {
+    try {
+      // Attempt to add newly created object to content admins group
```
Comment (elv-serban) [inline src/clients/Fabric.js:1093]: This comment doesn't apply here - right?  We're just retrieving the address of the content admins group

File: src/clients/Fabric.js [hunk 2]
```diff
@@ -1088,6 +1088,35 @@ const Fabric = {
...
+
+      const userGroups = await client.Collection({collectionType: "accessGroups"});
+      if(!userGroups.find(address => client.utils.EqualAddress(address, contentAdminGroupAddress))) {
+        throw "Current user not a member of content admins group " + contentAdminGroupAddress;
+      }
+
+      return contentAdminGroupAddress;
+    } catch(error) {
+      // eslint-disable-next-line no-console
+      console.error("Error adding new object to content admins group:");
```
Comment (elv-serban) [inline src/clients/Fabric.js:1114]: Should it be `Error retrieving content admins group`?

File: src/clients/Fabric.js [hunk 3]
```diff
@@ -1088,6 +1088,35 @@ const Fabric = {
...
+        methodName: "groupsMapping",
+        methodArgs: ["content_admin", 0],
+        formatArguments: true,
+      });
+
+      if(!contentAdminGroupAddress) {
+        throw "Unable to determine content admins group address";
+      }
+
+      const userGroups = await client.Collection({collectionType: "accessGroups"});
```
Comment (elv-serban) [inline src/clients/Fabric.js:1106]: I don't think we need to worry about the user being member of the content admins group.
A user that is not a content admin should be able to make a new object and give permissions to the content admins.

The one thing we need to do in the future is stronger validation that this user is part of the tenant.  But right now we don't have the necessary pieces to do it just yet.

File: src/stores/Object.js [hunk 4]
```diff
@@ -95,6 +95,34 @@ class ObjectStore {
...
+      yield Fabric.AddContentObjectGroupPermission({
+        objectId,
+        groupAddress: contentAdminsGroupAddress,
+        permission: "see"
+      });
+      yield Fabric.AddContentObjectGroupPermission({
+        objectId,
+        groupAddress: contentAdminsGroupAddress,
+        permission: "access"
+      });
```
Comment (elv-serban) [inline src/stores/Object.js:112]: Here we only need to add Manage.  The 'access' and 'see' are subsumed by Manage.

(plus it's less transactions which is good)

Comment (elv-zenia) [reply src/stores/Object.js:112]: @elv-serban I thought that was only true for library group permissions?

File: src/clients/Fabric.js (hunk 1 above)
Comment (elv-kevin) [reply src/clients/Fabric.js:1093]: yeah, I moved some stuff around after I wrote this, originally contained the actual permission grant

File: src/clients/Fabric.js (hunk 3 above)
Comment (elv-kevin) [reply src/clients/Fabric.js:1106]: you don't need to be a member of the group to grant permissions? ok

File: src/stores/Object.js (hunk 4 above)
Comment (elv-kevin) [reply src/stores/Object.js:112]: I checked with zenia and she said you need the 3. I'm fine with only adding manage if that works

Comment (elv-zenia) [reply src/stores/Object.js:112]: Ah nevermind, fabric browser does only add the one permission in that case

**Pull request merged**